components/script_bindings/webidls/CredentialsContainer.webidl

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+/*
+ * The origin of this IDL file is
+ * https://www.w3.org/TR/credential-management-1/#framework-credential-management
+ */
+
+// https://www.w3.org/TR/credential-management-1/#framework-credential-management
+partial interface Navigator {
+  [SecureContext, SameObject, Pref="dom_credential_management_enabled"] readonly attribute CredentialsContainer credentials;
+};
+
+// https://www.w3.org/TR/credential-management-1/#credentialscontainer
+[Pref="dom_credential_management_enabled", Exposed=Window, SecureContext]
+interface CredentialsContainer {
+  [Throws] Promise<Credential?> get(optional CredentialRequestOptions options = {});
+  [Throws] Promise<undefined> store(Credential credential);
+  [Throws] Promise<Credential?> create(optional CredentialCreationOptions options = {});
+  [Throws] Promise<undefined> preventSilentAccess();
+};
+
+// https://www.w3.org/TR/credential-management-1/#credentialrequestoptions-dictionary
+dictionary CredentialRequestOptions {
+  CredentialMediationRequirement mediation = "optional";
+  AbortSignal signal;
+};
+
+// https://www.w3.org/TR/credential-management-1/#dictdef-credentialcreationoptions
+dictionary CredentialCreationOptions {
```

Comment:
This dictionary is repeated. I suspect that's the source of the error.

Reply:
The repeat is a partial dictionary. I see that there is an issue for this already: #3588.

Reply:
CredentialRequestOptions needs the same treatment.

Reply:
Seems to be working now.